### ISSUE TITLE
Problem with ideal case tutorial

### ISSUE THREAD

**Issue by jpatiani** [scale-met/scale#4]
Dear @seiya san,

I have successfully installed scale v5.4.5 in a MacBook Pro with M1 chip, regardless of error pedantic argument mismatch (fixed with removed -pedantic-errors and add -fallow-argument-mismatch option). 

However, when I try to run the tutorial for ideal case it stops with the following error:

```
❯ mpirun -np 2 ./scale-rm_init init_R20kmDX500m.conf
 *** Start Launch System for SCALE-RM
 *** Execute preprocess? :  T
 *** Execute model?      :  F
ERROR  [FILE_def_axis] universal=000000, local=000000, jobID=000000, domain=000001
on rank 000000:  failed to define axis
Error: at line 1320 in file/scale_file_netcdf.c
       NetCDF: Name contains illegal characters
ERROR  [FILE_def_axis] universal=000001, local=000001, jobID=000000, domain=000001
on rank 000001:  failed to define axis
Error: at line 1320 in file/scale_file_netcdf.c
       NetCDF: Name contains illegal characters
 +++++ BULK   ID       :            0
 +++++ DOMAIN ID       :            1
 +++++ MASTER LOCATION :            0 /           2
 +++++ GLOBAL LOCATION :            0 /           2
 +++++ LOCAL  LOCATION :            0 /           2



Abort(-1) on node 1 (rank 1 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, -1) - process 1
Abort(-1) on node 0 (rank 0 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, -1) - process 0
```

I used configuration for MacOSX-gnu-ompi, although I used mpich. 
What can possibly be wrong with the compilation process?

Here is my netcdf config from nc-config and nf-config.

```
❯ nc-config --all

This netCDF 4.9.2 has been built with the following features:

  --cc            -> gcc
  --cflags        -> -I/usr/local/include -I/usr/local/include
  --libs          -> -L/usr/local/lib -lnetcdf
  --static        -> -lhdf5_hl -lhdf5 -lm -lz -lzip -lsz -lbz2 -lzstd -lxml2 -lcurl
  --has-dap          -> yes
  --has-dap2         -> yes
  --has-dap4         -> yes
  --has-nc2          -> yes
  --has-nc4          -> yes
  --has-hdf5         -> yes
  --has-hdf4         -> no
  --has-logging      -> no
  --has-pnetcdf      -> no
  --has-szlib        -> yes
  --has-cdf5         -> yes
  --has-parallel4    -> yes
  --has-parallel     -> yes
  --has-nczarr       -> yes
  --has-zstd         -> yes
  --has-benchmarks   -> no
  --has-multifilters -> yes
  --has-stdfilters   -> deflate bz2 zstd
  --has-quantize     -> yes

  --prefix           -> /usr/local
  --includedir       -> /usr/local/include
  --libdir           -> /usr/local/lib
  --plugindir        ->
  --version          -> netCDF 4.9.2

❯ nf-config --all

This netCDF-Fortran 4.6.0 has been built with the following features:

  --cc        -> gcc
  --cflags    -> -I/usr/local/include -I/usr/local/include

  --fc        -> gfortran
  --fflags    -> -I/usr/local/include -I/usr/local/include
  --flibs     -> -L/usr/local/lib -lnetcdff -L/usr/local/lib -lnetcdf -lnetcdf -lm -lz -lmpi
  --has-f90   ->
  --has-f03   -> yes

  --has-nc2   -> yes
  --has-nc4   -> yes

  --prefix    -> /usr/local
  --includedir-> /usr/local/include
  --version   -> netCDF-Fortran 4.6.0

```
I really appreciate any help you can provide.

**Comment by seiya** [comment]
Thank you for your report.
I don't have a M1 Mac, so I am not sure what the cause is, but I think it is due to the version of the GNU compiler.
It seems to be caused by an incorrect data transfer from fortran to C.

SCALE 5.4.x will not compile on gfortran version 10 or higher (see https://github.com/scale-met/scale/wiki/Information-about-released-version) due to the argument mismatch.
The development version fixes this by using ISO_C_BINDING.

The next release with this fix will be available in a few months.

**Comment by jpatiani** [comment]
Dear @seiya san,
Thank you for getting back to me.
I forgot to report what compiler I used in the first message. I think you're correct. I use gcc gfortran version 12.2. Also, I was late to notice the information about the gfortran compiler issue before I post the message.

Anyway, thank you very much for your help.

I think there are no issues regarding compiling in M1 Mac except for the type mismatch error. Hopefully, the fix will come out soon.

Best regards.

**Comment by seiya** [comment]
We have released a new version 5.5.0.
The web has not been updated yet, but you can get it from github.
We expect the error to be resolved in this version.

**Comment by jpatiani** [comment]
> We have released a new version 5.5.0. The web has not been updated yet, but you can get it from github. We expect the error to be resolved in this version.

Thank you very much for your notification. I will try to install the new version.

Best regards

**Comment by jpatiani** [comment]
@seiya, MPI_Win_allocate in ommunication/scale_comm_cartesC.F90 gave me errors

`communication/scale_comm_cartesC.F90:516:89:`
`516 |                                 ginfo(gid)%recvbuf_WE(n), ginfo(gid)%win_packWE(n), ierr)`
`.      |                                                                                         1`
`Error: Type mismatch in argument ‘baseptr’ at (1); passed TYPE(c_ptr) to INTEGER(8)
communication/scale_comm_cartesC.F90:534:89:`
`534 |                                 ginfo(gid)%recvbuf_NS(n), ginfo(gid)%win_packNS(n), ierr)`
`         |                                                                                         1`
`Error: Type mismatch in argument ‘baseptr’ at (1); passed TYPE(c_ptr) to INTEGER(8)`
  
Even if I use fallow_argument_mismatch, that doesn't help to keep off the error.

Best regards.

**Comment by seiya** [comment]
@jpatiani Thank you for the report.
I tried to correct it.
However, since the error is not reproduced in the environment at hand, could you please try to see if it has been corrected?
branch: hotfix/5.5.1 https://github.com/scale-met/scale/tree/hotfix/5.5.1

**Comment by jpatiani** [comment]
> @jpatiani Thank you for the report. I tried to correct it. However, since the error is not reproduced in the environment at hand, could you please try to see if it has been corrected? branch: hotfix/5.5.1 https://github.com/scale-met/scale/tree/hotfix/5.5.1

@seiya, 
Thank you very much. I have successfully compiled and run the ideal case tutorial.
I will check the result to verify the output.

Best regards.

**Comment by seiya** [comment]
@jpatiani Sorry for the late reply.
Fixed version 5.5.1 has been released.
